ddtrace/tracer/span.go

```diff
@@ -423,7 +448,7 @@ func (s *span) setTagBool(key string, v bool) {
 
 // setMetric sets a numeric tag, in our case called a metric. This method
 // is not safe for concurrent use.
-func (s *span) setMetric(key string, v float64) {
+func (s *Span) setMetric(key string, v float64) {
 	if s.Metrics == nil {
 		s.Metrics = make(map[string]float64, 1)
 	}
```

Comment:
This is one more occurrence of a method that is unsafe for a `nil` receiver when everything else is... This is confusing to me (and probably a 🦶🏻🔫).

Reply:
I've been debating. My thought was that only exported methods should have a `nil` check and internal ones should be used carefully by us, but perhaps it's a cheap enough check that we should do it everywhere to be safe.

WDYT?